src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/VariantsToTable.java

```diff
@@ -0,0 +1,430 @@
+     * unreadable/malformed for certain tools like R, as the representation of multi-allelic INFO field values are often comma-separated lists
+     * of values.  Using the flag will cause multi-allelic records to be split into multiple lines of output (one for each allele in the ALT field);
+     * INFO field values that are not lists are copied for each of the output records while only the appropriate entry is used for lists.
+     */
+    @Argument(fullName="splitMultiAllelic", shortName="SMA", doc="If provided, we will split multi-allelic records into multiple lines of output", optional=true)
+    private boolean splitMultiAllelic = false;
+
+    /**
+     * By default, this tool emits one line per usable VCF record (or per allele if the -SMA flag is provided).  Using the -moltenize flag
+     * will cause records to be split into multiple lines of output: one for each field provided with -F or one for each combination of sample
+     * and field provided with -GF.  Note that the "Sample" column for -F fields will always be "site".
+     */
+    @Advanced
+    @Argument(fullName="moltenize", shortName="moltenize", doc="If provided, we will produce molten output", optional=true)
+    private boolean moltenizeOutput = false;
+
+    /**
+     * By default, this tool throws a UserException when it encounters a field without a value in some record.  This
+     * is generally useful when you mistype -F CHROM, so that you get a friendly warning about CHROM not being
+     * found before the tool runs through 40M 1000G records.  However, in some cases you genuinely want to allow such
+     * fields (e.g., AC not being calculated for filtered records, if included).  When provided, this argument
+     * will cause VariantsToTable to write out NA values for missing fields instead of throwing an error.
+     */
+    @Advanced
+    @Argument(fullName="allowMissingData", shortName="AMD", doc="If provided, we will not require every record to contain every field", optional=true)
+    private boolean ALLOW_MISSING_DATA = false;
+    private static final String MISSING_DATA = "NA";
+
+
+    private final List<String> samples = new ArrayList<>();
+    private long nRecords = 0L;
+    private PrintStream outputStream = null;
+
+    @Override
+    public void onTraversalStart() {
+        outputStream = createPrintStream();
+
+        if ( !genotypeFieldsToTake.isEmpty() ) {
+            final Map<String, VCFHeader> vcfHeaders = Collections.singletonMap(getDrivingVariantsFeatureInput().getName(), getHeaderForVariants());
+            final SortedSet<String> vcfSamples = new TreeSet<>(VcfUtils.getSampleList(vcfHeaders, GATKVariantContextUtils.GenotypeMergeType.REQUIRE_UNIQUE));
+            samples.addAll(vcfSamples);
+
+            // optimization: if there are no samples, we don't have to worry about any genotype fields
+            if ( samples.isEmpty() ) {
+                genotypeFieldsToTake.clear();
```

Comment:
If we get to this line we should log a warning saying why we're ignoring the genotype fields, and possibly throw if fieldsToTake is empty since otherwise we're about to chug through a lot of data for no reason.